src/markdown.c

```diff
@@ -1413,6 +1413,24 @@ prefix_quote(uint8_t *data, size_t size)
 	return 0;
 }
 
+static size_t
+prefix_blockspoiler(uint8_t *data, size_t size)
+{
+        size_t i = 0;
+        if (i < size && data[i] == ' ') i++;
+        if (i < size && data[i] == ' ') i++;
+        if (i < size && data[i] == ' ') i++;
+
+        if ((i < size && data[i] == '>') && (i < size && data[i + 1] == '!')) {
```

Comment:
Looks like `i < size && data[i + 1] == '!'` might allow an out-of-bounds read? I don't believe `data` is guaranteed to be null-terminated until `bufcstr()` is called either, so there could be minor weirdness here.

This expression could be simplified to `i + 1 < size && data[i] == '>' && data[i + 1] == '!'`.